From d4e37496e85b68ce1288ea57c05acd4a1854d938 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Thu, 11 Feb 2010 14:28:58 +1000
Patch-mainline: 2.6.34
References: fate#310916
Git-commit: 8dff4742a6a718e71d15a9b7a8d3d2255bfbc91a
Subject: [PATCH 0370/2588] drm/kms: fix fb_changed = true else statement

a patch from Roel was wrong, fix this properly, really
if the fb ptrs are different fb changed shuold be true.

Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit 8dff4742a6a718e71d15a9b7a8d3d2255bfbc91a)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/drm_crtc_helper.c |    6 +-----
 1 files changed, 1 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/drm_crtc_helper.c b/drivers/gpu/drm/drm_crtc_helper.c
index ec8cfd6..220177a 100644
--- a/drivers/gpu/drm/drm_crtc_helper.c
+++ b/drivers/gpu/drm/drm_crtc_helper.c
@@ -837,11 +837,7 @@ int drm_crtc_helper_set_config(struct drm_mode_set *set)
 			mode_changed = true;
 		} else if (set->fb == NULL) {
 			mode_changed = true;
-		} else if ((set->fb->bits_per_pixel !=
-			 set->crtc->fb->bits_per_pixel) ||
-			 set->fb->depth != set->crtc->fb->depth)
-			fb_changed = true;
-		else
+		} else
 			fb_changed = true;
 	}
 
-- 
1.7.6

